### PR TITLE
Multiple HKG car ports

### PR DESCRIPTION
**Pull request by xps-genesis**
Added below vehicle models-

KONA_HEV = "HYUNDAI KONA HEV 2019"
SONATA_H = "HYUNDAI SONATA HYBRID 2020"
GRANDEUR = "GRANDEUR IG 2017"
GRANDEUR_HEV = "GRANDEUR IG HEV 2019"
KIA_NIRO_EV = "KIA NIRO EV 2020 PLATINUM"
KIA_NIRO_HEV = "KIA NIRO HEV 2018"
KIA_CEED = "KIA CEED 2019"
KIA_SPORTAGE = "KIA SPORTAGE S 2020"
KIA_CARDENZA = "KIA K7 2016-2019"
KIA_CARDENZA_HEV = "KIA K7 HEV 2016-2019"

cleaned up multiple fingerprint strings for a vehicle into 1.

marked non SCC fingerprints

### CODE REVIEW

**Comment by xps-genesis** [comment]
@adeebshihadeh : My fear/doubt has come true. the unit tests failed because the fingerprint string of the KIA version of the same Hyundai car is same or has subset of the MSG string. So the fingerprinting logic fails to identify between the 2 vehicles.

>> Those two fingerprints are inconsistent HYUNDAI KONA HEV 2019 KIA NIRO HEV 2018
>> Those two fingerprints are inconsistent HYUNDAI IONIQ HYBRID PREMIUM 2018-2020 KIA NIRO HEV 2018

how do we tackle this. just call HKG_KONA_NIRO_HEV and HKG_IONIQ_NIRO_HEV and merge the strings?

>> Those two fingerprints are inconsistent KIA OPTIMA SX 2019 & 2016 KIA SPORTAGE S 2020 (these both are non scc fingerprints, probably good to remove support as of now)

**Comment by pd0wm** [comment]
Just support the Kia Niro with fingerprinting 2.0 only. Create an array like this https://github.com/commaai/openpilot/blob/master/selfdrive/car/toyota/values.py#L272, with the Kia Niro and the tests will pass.

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -115,35 +94,62 @@ class Buttons:
...
+  }],
+  CAR.GRANDEUR_HEV: [{
+    68: 8, 127: 8, 304: 8, 320: 8, 339: 8, 352: 8, 356: 4, 516: 8, 544: 8, 546: 8, 576: 8, 593: 8, 688: 5, 832: 8, 865: 8, 881: 8, 882: 8, 897: 8, 902: 8, 903: 8, 905: 8, 909: 8, 913: 8, 916: 8, 1040: 8, 1042: 8, 1056: 8, 1057: 8, 1078: 4, 1108: 8, 1136: 6, 1138: 5, 1151: 8, 1155: 8, 1156: 8, 1157: 4, 1162: 8, 1164: 8, 1168: 8, 1173: 8, 1180: 8, 1185: 8, 1186: 2, 1191: 2, 1193: 8, 1210: 8, 1225: 8, 1227: 8, 1265: 4, 1268: 8, 1280: 1, 1287: 4, 1290: 8, 1291: 8, 1292: 8, 1294: 8, 1312: 8, 1322: 8, 1342: 8, 1345: 8, 1348: 8, 1355: 8, 1363: 8, 1369: 8, 1371: 8, 1378: 8, 1407: 8, 1419: 8, 1425: 2, 1427: 6, 1429: 8, 1430: 8, 1448: 8, 1456: 4, 1470: 8, 1476: 8, 1535: 8
+  }],
+  CAR.KIA_SPORTAGE: [{ # no SCC msgs in this string
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/values.py:126]: Let's remove any models we don't have an SCC fingerprint for. We can add them back with a fingerprint from only cars with SCC.

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -115,35 +94,62 @@ class Buttons:
...
+  }],
   CAR.KONA_EV: [{
     127: 8, 304: 8, 320: 8, 339: 8, 352: 8, 356: 4, 544: 8, 549: 8, 593: 8, 688: 5, 832: 8, 881: 8, 882: 8, 897: 8, 902: 8, 903: 8, 905: 8, 909: 8, 916: 8, 1040: 8, 1042: 8, 1056: 8, 1057: 8, 1078: 4, 1136: 8, 1151: 6, 1168: 7, 1173: 8, 1183: 8, 1186: 2, 1191: 2, 1225: 8, 1265: 4, 1280: 1, 1287: 4, 1290: 8, 1291: 8, 1292: 8, 1294: 8, 1307: 8, 1312: 8, 1322: 8, 1342: 6, 1345: 8, 1348: 8, 1355: 8, 1363: 8, 1369: 8, 1378: 4, 1407: 8, 1419: 8, 1426: 8, 1427: 6, 1429: 8, 1430: 8, 1456: 4, 1470: 8, 1473: 8, 1507: 8, 1535: 8, 2000: 8, 2004: 8, 2008: 8, 2012: 8
   }],
-  CAR.KIA_FORTE: [{
+  CAR.KIA_FORTE: [{ # no SCC msgs in this string
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/values.py:108]: remove non-scc

File: selfdrive/car/hyundai/values.py [hunk 3]
```diff
@@ -115,35 +94,62 @@ class Buttons:
...
+  CAR.KIA_CADENZA_HEV: [{
+    68: 8, 127: 8, 304: 8, 320: 8, 339: 8, 352: 8, 356: 4, 544: 8, 576: 8, 593: 8, 688: 5, 832: 8, 865: 8, 881: 8, 882: 8, 897: 8, 902: 8, 903: 8, 905: 8, 909: 8, 913: 8, 916: 8, 1040: 8, 1042: 8, 1056: 8, 1057: 8, 1078: 4, 1096: 8, 1102: 8, 1108: 8, 1136: 6, 1138: 5, 1151: 8, 1155: 8, 1156: 8, 1157: 4, 1162: 8, 1164: 8, 1168: 7, 1173: 8, 1180: 8, 1186: 2, 1191: 2, 1210: 8, 1227: 8, 1265: 4, 1268: 8, 1280: 1, 1287: 4, 1290: 8, 1291: 8, 1292: 8, 1294: 8, 1312: 8, 1322: 8, 1342: 6, 1343: 8, 1345: 8, 1348: 8, 1355: 8, 1363: 8, 1369: 8, 1371: 8, 1378: 8, 1379: 8, 1407: 8, 1419: 8, 1427: 6, 1429: 8, 1430: 8, 1448: 8, 1456: 4, 1470: 8, 1476: 8, 1535: 8
+  }],
 }
 
 ECU_FINGERPRINT = {
   Ecu.fwdCamera: [832, 1156, 1191, 1342]
 }
 
+# Don't use theses fingerprints for fingerprinting, they are still needed for ECU detection
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/values.py:150]: typo

File: selfdrive/test/test_car_models.py [hunk 4]
```diff
@@ -426,6 +426,16 @@ def get_route_log(route_name):
...
+  HYUNDAI.KONA_HEV,
+  HYUNDAI.GRANDEUR,
+  HYUNDAI.GRANDEUR_HEV,
+  HYUNDAI.KIA_NIRO_EV,
+  HYUNDAI.KIA_NIRO_HEV,
+  HYUNDAI.SONATA_H,
+  HYUNDAI.KIA_CADENZA,
+  HYUNDAI.KIA_CEED,
+  HYUNDAI.KIA_CADENZA_HEV,
+  HYUNDAI.KIA_SPORTAGE,
```
Comment (adeebshihadeh) [inline selfdrive/test/test_car_models.py:438]: Let's not add anything to the non-tested list. New models will require a test route before merging.

**Comment by adeebshihadeh** [comment]
Can you rebase so that the new tests run?

**Comment by xps-genesis** [comment]
added new fingerprint vehicles to issue #1929 for routes

**Comment by adeebshihadeh** [comment]
Thanks for the PR @xps-genesis, closing this as I think it'll be easier to do individual PRs as we get routes for these cars.

**Pull request closed without merging**